src/main/scala/gitbucket/core/model/activity/comment/CommentActivityInfo.scala

```diff
@@ -0,0 +1,95 @@
+package gitbucket.core.model.activity.comment
+
+import java.util.UUID
+
+import gitbucket.core.model.Activity
+import gitbucket.core.model.Profile.currentDate
+import gitbucket.core.model.activity.BaseActivityInfo
+
+sealed trait CommentActivityInfo extends BaseActivityInfo {
+  def userName: String
+  def repositoryName: String
+  def activityUserName: String
+  def comment: String
+  def maybeIssueId: Option[Int]
+  def maybeCommitId: Option[String]
+
+  protected final def cut(value: String, length: Int): String =
+    if (value.length > length) s"${value.substring(0, length)}..."
+    else value
+}
+
+final case class IssueCommentInfo(
+  userName: String,
+  repositoryName: String,
+  activityUserName: String,
+  comment: String,
+  issueId: Int
+) extends CommentActivityInfo {
+
+  override def maybeIssueId: Option[Int] = Some(issueId)
```

Comment:
In reference to [your comment](https://github.com/gitbucket/gitbucket/pull/2526/files/18e3955e5b496d541220b72ae1b02a37d694688d#r478452455), a overrided value is provided here

Reply:
My question was who uses this value? Is this also for future use?

Reply:
Sorry for not addressing your question properly; this value is necessary for the `CommentActivityInfo` trait to be subtyped by both `IssueCommentInfo` and `CommitCommentInfo`. 

I see that you've also asked a question regarding why the intermediate trait is necessary, and I've provided an explanation [here](https://github.com/gitbucket/gitbucket/pull/2526#discussion_r478523839). In my opinion, this is a small sacrifice in order to have the benefits of exhaustive pattern-matching and compile-time checks

Reply:
I understand your intention of introducing intermediate traits. Regarding these `maybeXxxx` methods, since no one calls these methods currently, so I thought these methods are defined for future use as same as intermediate traits. Is my understanding correct?

My personal preference is adding something when we need them. Therefore, these methods (and intermediate traits) looks redundant for me at the moment, to be honest.

Reply:
Thanks for the feedback; I guess we have different ideas of how this should look. I'll defer to you since you have many more years of experience with this codebase than I have.